fineract-provider/src/main/java/org/apache/fineract/useradministration/service/UserDataValidator.java

```diff
@@ -75,6 +80,14 @@ public void validateForCreate(final String json) {
         final String username = this.fromApiJsonHelper.extractStringNamed("username", element);
         baseDataValidator.reset().parameter("username").value(username).notBlank().notExceedingLengthOf(100);
 
+        if (username != null) {
```

Comment:
I believe there is no point going inside this code block, if the username is blank/empty.
```suggestion
        if (StringUtils.isEmpty(username)) {
```
```suggestion
        if (StringUtils.isBlank(username)) {
```

Reply:
That makes sense. I've applied the optimization suggestion @mohitsinha